In [13]:
import pandas as pd
import os
import seaborn as sns
import  matplotlib.pyplot as plt
from pathlib import Path

In [4]:
df = pd.read_csv("hf://datasets/dazzle-nu/CIS435-CreditCardFraudDetection/fraudTrain.csv")
df.head()

/Users/romacarapetean/Desktop/Projects/Credit_Card_Fraud_Detection/.venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Unnamed: 0 trans_date_trans_time        cc_num  \
0           0           1/1/19 0:00  2.703190e+15   
1           1           1/1/19 0:00  6.304230e+11   
2           2           1/1/19 0:00  3.885950e+13   
3           3           1/1/19 0:01  3.534090e+15   
4           4           1/1/19 0:03  3.755340e+14   

                             merchant       category     amt      first  \
0          fraud_Rippin, Kub and Mann       misc_net    4.97   Jennifer   
1     fraud_Heller, Gutmann and Zieme    grocery_pos  107.23  Stephanie   
2                fraud_Lind-Buckridge  entertainment  220.11     Edward   
3  fraud_Kutch, Hermiston and Farrell  gas_transport   45.00     Jeremy   
4                 fraud_Keeling-Crist       misc_pos   41.96      Tyler   

      last gender                        street  ... city_pop  \
0    Banks      F                561 Perry Cove  ...     3495   
1     Gill      F  43039 Riley Greens Suite 393  ...      149   
2  Sanchez      M      594 White Dale Suite 530  ...     4154   
3    White      M   9443 Cynthia Court Apt. 038  ...     1939   
4   Garcia      M              408 Bradley Rest  ...       99   

                                 job      dob  \
0          Psychologist, counselling   3/9/88   
1  Special educational needs teacher  6/21/78   
2        Nature conservation officer  1/19/62   
3                    Patent attorney  1/12/67   
4     Dance movement psychotherapist  3/28/86   

                          trans_num   unix_time  merch_lat  merch_long  \
0  0b242abb623afc578575680df30655b9  1325376018  36.011293  -82.048315   
1  1f76529f8574734946361c461b024d99  1325376044  49.159047 -118.186462   
2  a1a22d70485983eac12b5b88dad1cf95  1325376051  43.150704 -112.154481   
3  6b849c168bdad6f867558c3793159a81  1325376076  47.034331 -112.561071   
4  a41d7549acf90789359a9aa5346dcb46  1325376186  38.674999  -78.632459   

  is_fraud Unnamed: 23  6006  
0        0         NaN   NaN  
1        0         NaN   NaN  
2        0         NaN   NaN  
3        0         NaN   NaN  
4        0         NaN   NaN  

[5 rows x 25 columns]

In [5]:
print(df.columns)

Index(['Unnamed: 0', 'trans_date_trans_time', 'cc_num', 'merchant', 'category',
       'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip',
       'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'unix_time',
       'merch_lat', 'merch_long', 'is_fraud', 'Unnamed: 23', '6006'],
      dtype='object')


In [7]:
null_counts = df.isnull().sum()
print(null_counts)

Unnamed: 0                     0
trans_date_trans_time          0
cc_num                         0
merchant                       0
category                       0
amt                            0
first                          0
last                           0
gender                         0
street                         0
city                           0
state                          0
zip                            0
lat                            0
long                           0
city_pop                       0
job                            0
dob                            0
trans_num                      0
unix_time                      0
merch_lat                      0
merch_long                     0
is_fraud                       0
Unnamed: 23              1048575
6006                     1048575
dtype: int64


The dataset doesn't have any null values, however it has Personal Data columns like and unnamed columns:
'cc_num', 'trans_num', 'first', 'last', 'street', 'Unnamed: 0', '6006', 'Unnamed: 23' which are gonna be deleted in the preprocessing step

In [43]:
# Read the preprocessed DataFrame for further visualization
cleaned_path = Path('../data_cache/cleanedFraudDataset.csv')
preprocessed_df = pd.read_csv(cleaned_path)
preprocessed_df.head()

category     amt gender            city state    zip      lat  \
0       misc_net    4.97      F  Moravian Falls    NC  28654  36.0788   
1    grocery_pos  107.23      F          Orient    WA  99160  48.8878   
2  entertainment  220.11      M      Malad City    ID  83252  42.1808   
3  gas_transport   45.00      M         Boulder    MT  59632  46.2306   
4       misc_pos   41.96      M        Doe Hill    VA  24433  38.4207   

       long  city_pop                                job   unix_time  \
0  -81.1781      3495          Psychologist, counselling  1325376018   
1 -118.2105       149  Special educational needs teacher  1325376044   
2 -112.2620      4154        Nature conservation officer  1325376051   
3 -112.1138      1939                    Patent attorney  1325376076   
4  -79.4629        99     Dance movement psychotherapist  1325376186   

   merch_lat  merch_long  is_fraud  merchant_encoding  age  
0  36.011293  -82.048315         0                514   31  
1  49.159047 -118.186462         0                241   41  
2  43.150704 -112.154481         0                390   57  
3  47.034331 -112.561071         0                360   52  
4  38.674999  -78.632459         0                297   33